Automattic/wp-calypso
# Add different comparison plans for guided onboarding

Related to 7377-gh-Automattic/dotcom-forge
## Proposed Changes
We need to show a subset of plans based on segmentation but show all the plans in the comparison grid.

https://github.com/Automattic/dotcom-forge/assets/7931304/f95d1f44-b256-4b36-bce9-8ae78bf2452d

* This PR introduces `usePlansForComparison` which gets passed to `usePlanTypesWithIntent` to return the appropriate plans for the comparison table for the relevant case.

## Why are these changes being made?



## Testing Instructions

1. Go to /start/guided
2. Trigger the relevant segments by answering the questions below.
3. Ensure that the comparison table lists all the plans.

Intent (Q1) | Goals (Q2) | Segment | Plans or action
-- | -- | -- | --
Migrate or import | n/a | Migration | Redirect to import
Create for clients | n/a | Developer / Agency | Creator, Entrepreneur, Enterprise
Create for self | Unknown (question skipped) | Unknown | All plans
Create for self | includes Get a website built quickly | DIFM | Redircet to DIFM
Create for self | Sell something and not Get a website built quickly | Merchant | Creator, Entrepreneur, Enterprise
Create for self | not Get a website built quickly and not Sell something | Consumer / Business | Starter, Explorer, Creator
Create for self | Publish a blog | Blogger | Free, Starter, Explorer, Creator
Create for self | Educational/nonprofit | Nonprofit | Free, Starter, Explorer, Creator

4. Test for possible regressions




## Pre-merge Checklist
